core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -913,6 +939,18 @@ object IO extends Serializable {
       )
     )
 
+  /**
+   * The moral equivalent of `if (p) exp`
+   */
+  final def when[E](b: Boolean)(io: IO[E, Unit]): IO[E, Unit] =
+    if (b) io else IO.unit
+
+  /**
+   * The moral equivalent of `if (p) exp` when `p` has side-effects
+   */
+  final def whenM[E](b: IO[Nothing, Boolean])(io: IO[E, Unit]): IO[E, Unit] =
```

Comment:
You could (should?) add syntax for these in the syntax module.

Reply:
Like this?
```
io.when(pred)
```
Then I may just add `io.when` directly. Somehow the opposite (`pred.when(io)`) feels weird.

Reply:
ha no, only on `IO[E, Unit]`